src/sentry/static/sentry/app/components/projectHeader/index.jsx

```diff
@@ -37,8 +37,9 @@ class ProjectHeader extends React.Component {
 
     // TODO: remove when feature is released
     let hasEnvironmentsFeature = new Set(org.features).has('environments');
-
-    let showEnvironmentsToggle = hasEnvironmentsFeature && navSection !== 'settings';
+    let pagesWithEnvironments = new Set(['stream', 'releases', 'dashboard']);
```

Comment:
Is this a better experience than always showing the toggle (despite data not changing)?

I think there'll be a slight mental overhead having to figure out which tab you have to navigate to in order to select an environment

Reply:
cc @saragilford @ckj

Reply:
@billyvg to my mind its more confusing if a user does select an environment, and the page is not responsive to that at all (ie, if they are looking for user feedback from their test env, they toggle into `test`, and the page doesn't change at all). Could also use input from @Chrissy here about what's the best way to handle.

Also @lynnagara @tkaemming -- what about the events page? It's in the spec as a page that should be updated with environment. Did this get scrapped?

Reply:
It was an oversight, adding to the list

Reply:
👍

Reply:
What about keeping it visible but disabled?

Reply:
Normally I prefer to gray things out or generally make them look disabled, because I think it is more communicative.

But if we are within a view where the filter is not filtering on any of the items in the view, then that is an exception and we should hide it.

Truthfully, this might be a little bit of a smell regarding the placement of the environment toggle (probably should feel more local), but that is for another day...